src/net/fe/network/command/MoveCommand.java

```diff
@@ -39,7 +37,8 @@ public Runnable applyClient(ClientOverworldStage stage, Unit unit, ArrayList<Att
 		
 		return new Runnable() {
 			public void run() {
-				Path p = stage.grid.getShortestPath(unit, unit.getXCoord()+moveX, unit.getYCoord()+moveY);
+				Path p = new Path();
+				p.setNodes(path);
 				stage.grid.move(unit, unit.getXCoord()+moveX, unit.getYCoord()+moveY, true);
```

Comment:
`unit.getXCoord()+moveX` should be the same as `path[path.length - 1].x`; similar for the y coordinate. Using that equivalency all four times, it should be possible to prune the `moveX` and `moveY` variables altogether.